File: OahuDemo/ViewController.m
//
//  ViewController.m
//  OahuDemo
//
//  Created by Miguel Bassila on 12/11/15.
//  Copyright © 2015 Miguel Bassila. All rights reserved.
//

#import "ViewController.h"
#import "Oahu-Swift.h"

@import WebKit;

@interface ViewController () <WKNavigationDelegate>

@end

@implementation ViewController

- (void)viewDidLoad
{
  [super viewDidLoad];
//  WKWebView *webView = [[WKWebView alloc] initWithFrame:self.view.frame];
//  [self.view addSubview:webView];
//  webView.navigationDelegate = self;
//
//  [webView loadRequest:[NSURLRequest requestWithURL:[NSURL URLWithString:@"http://www.uol.com.br"]]];
//

  Oahu *oahu = [[Oahu alloc] initForView:self.view allowsBackForwardNavigationGestures:YES];
  oahu.navigationDelegate = self;
  [oahu loadRequest:@"http://www.globo.com"];
}

-(void)webView:(WKWebView *)webView didCommitNavigation:(WKNavigation *)navigation
{
  NSLog(@"Ola");
}

@end

File: OahuDemo/ViewController.swift
import UIKit
import oahu
import WebKit

class ViewController: UIViewController, OahuDelegate {
    var oahu: Oahu?

    var interceptor: Interceptor?
    var contentType: String?

    override func viewDidLoad() {
        super.viewDidLoad()

        oahu = Oahu(forView: view, allowsBackForwardNavigationGestures: true, interceptor: interceptor)
        oahu?.loadRequest("http://api.openweathermap.org/data/2.5/weather?q=London,uk&appid=2de143494c0b295cca9337e1e96b00e0")
//        oahu?.loadRequest("http://www.elo7.com.br")
        oahu?.oahuDelegate = self
    }


    func webView(webView: WKWebView, didFinishNavigation navigation: WKNavigation!) {
        if let contentType = self.contentType  {
            if contentType == "application/json; charset=utf-8" {
                let data = try! String(contentsOfURL: webView.URL!, encoding: NSUTF8StringEncoding)
                print(data)

                let dick = convertStringToDictionary(data)
                print(dick)
            }
        }
    }

    func convertStringToDictionary(text: String) -> [String: AnyObject]? {
        if let data = text.dataUsingEncoding(NSUTF8StringEncoding) {
            do {
                return try NSJSONSerialization.JSONObjectWithData(data, options: NSJSONReadingOptions()) as? [String:AnyObject]
            } catch {
                return nil
            }
        }
        return nil
    }

    func webView(webView: WKWebView, decidePolicyForNavigationResponse navigationResponse: WKNavigationResponse, decisionHandler: (WKNavigationResponsePolicy) -> Void) {
        if let response = navigationResponse.response as? NSHTTPURLResponse {
            if let contentType = response.allHeaderFields["Content-Type"] as? String {
                self.contentType = contentType
            }
        }

        decisionHandler(.Allow)
    }
}



